wlauto/workloads/googleslides/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java

```diff
@@ -0,0 +1,469 @@
+            startDumpsys(ACTIVITY_SLIDES);
+            slideTimer = new Timer();
+            slideTimer.start();
+            uiDeviceSwipeHorizontal(centerX - centerX/2, centerX + centerX/2, centerY);
+            slideTimer.end();
+            results.put(testTag, slideTimer);
+            endDumpsys(ACTIVITY_SLIDES, testTag);
+            sleep(1);
+        }
+        timer.end();
+        results.put("slides_reverse", timer);
+        endDumpsys(ACTIVITY_SLIDES, "slides_reverse");
+        sleep(1);
+
+        // scroll forward in slideshow mode
+        timer = new Timer();
+        timer.start();
+        clickView(BY_DESC, "Start slideshow", true);
+        timer.end();
+        results.put("open_slideshow", timer);
+
+        startDumpsys(ACTIVITY_SLIDES);
+        timer = new Timer();
+        timer.start();
+        while (++slideIndex < slideCount) {
+            testTag = "slideshow_next_" + slideIndex;
+            startDumpsys(ACTIVITY_SLIDES);
+            slideTimer = new Timer();
+            slideTimer.start();
+            uiDeviceSwipeHorizontal(centerX + centerX/2, centerX - centerX/2, centerY);
+            slideTimer.end();
+            results.put(testTag, slideTimer);
+            endDumpsys(ACTIVITY_SLIDES, testTag);
+            sleep(1);
+        }
+        timer.end();
+        results.put("play_slideshow", timer);
+        endDumpsys(ACTIVITY_SLIDES, "play_slideshow");
+        sleep(1);
+
+        getUiDevice().pressBack();
+        getUiDevice().pressBack();
+    }
+
+    protected void testEditNewSlidesDocument(String docName) throws Exception {
+        startDumpsys(ACTIVITY_DOCLIST);
+        // create new file
+        timer = new Timer();
+        timer.start();
+        clickView(BY_DESC, "New presentation");
+        clickView(BY_TEXT, "New PowerPoint", true);
+        timer.end();
+        results.put("create_document", timer);
+        endDumpsys(ACTIVITY_DOCLIST, "create_document");
+
+        // first slide
+        enterTextInSlide("Title", "WORKLOAD AUTOMATION");
+        enterTextInSlide("Subtitle", "Measuring perfomance of different productivity apps on Android OS");
+        saveDocument(docName);
+
+        insertSlide("Title and Content");
+        enterTextInSlide("title", "Extensions - Workloads");
+        enterTextInSlide("Text placeholder", SLIDE_TEXT_CONTENT);
+        clickView(BY_DESC, "Text placeholder");
+        clickView(BY_DESC, "Format");
+        clickView(BY_TEXT, "Droid Sans");
+        clickView(BY_TEXT, "Droid Sans Mono");
+        clickView(BY_ID, PACKAGE_ID + "palette_back_button");
+        UiObject decreaseFont = getViewByDesc("Decrease text");
+        repeatClickView(decreaseFont, 20);
+        getUiDevice().pressBack();
+
+        // get image from gallery and insert
+        insertSlide("Title Only");
+        clickView(BY_DESC, "Insert");
+        clickView(BY_TEXT, "Image", true);
+        clickView(BY_TEXT, "Recent");
+        try {
```

Comment:
As with the default folder, to keep the test simple we just select the most recent image from the gallery (regardless of what folder it is located in). It performed more reliably than pushing an image then trying to find it in the picker.
The image did not seem to affect the results, and in the rare case that there will be none at all, the test does not fall over.